File: Ser2023/Company/Company.Web/Modules/ServerTypes/Store/WaresOrderDetailForm.ts
import { IntegerEditor, DecimalEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface WaresOrderDetailForm {
    WaresID: IntegerEditor;
    SinglePrice: DecimalEditor;
    Quantity: DecimalEditor;
    Discount: DecimalEditor;
}

export class WaresOrderDetailForm extends PrefixedContext {
    static formKey = 'Store.WaresOrderDetails';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!WaresOrderDetailForm.init)  {
            WaresOrderDetailForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DecimalEditor;

            initFormType(WaresOrderDetailForm, [
                'WaresID', w0,
                'SinglePrice', w1,
                'Quantity', w1,
                'Discount', w1
            ]);
        }
    }
}

File: Ser2023/Company/Company.Web/Modules/Store/WaresOrderDetail/WaresOrderDetailForm.cs
using Serenity.ComponentModel;
using System;

namespace Company.Store.Forms;

[FormScript("Store.WaresOrderDetails")]
[BasedOnRow(typeof(WaresOrderDetailRow), CheckNames = true)]
public class WaresOrderDetailForm
{
    [EditLink, Width(200)]
    public Int32 WaresID { get; set; }
    [Width(100)]
    public Decimal SinglePrice { get; set; }
    [Width(100)]
    public Single Quantity { get; set; }
    [Width(100)]
    public Single Discount { get; set; }
}